### PR TITLE
WIP - CL support for Apache Thrift

### PR DESCRIPTION
**Pull request by uint**
The current state of work. Brief summary:

* Patched the head of Apache Thrift with the de.setf code.
* Fixed errors and warnings. Got it working with SBCL. This involves:
  * The IDL -> CL generator.
  * The CL library.
* Created the cross-test code and integrated it so it builds together with the rest of cross-tests. Mostly everything is implemented, with the exception of exceptions (pun unintended) on the client side - apparently receiving an exception as a response is broken right now.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -924,6 +938,7 @@ echo "Building Python Library ...... : $have_python"
 echo "Building Py3 Library ......... : $have_py3"
 echo "Building Ruby Library ........ : $have_ruby"
 echo "Building Rust Library ........ : $have_rs"
+echo "Preparing Common Lisp tests... : $have_cl"
```
Comment (dkochmanski) [inline configure.ac]: why isn't it `Building Common Lisp Library`?

File: lib/cl/READMES/readme-cassandra.lisp [hunk 2]
```diff
@@ -0,0 +1,66 @@
+;;; -*- Mode: lisp; Syntax: ansi-common-lisp; Base: 10; Package: cl-user; -*-
+
+(in-package :cl-user)
+
+#+(or ccl sbcl sbcl) /development/source/library/
```
Comment (dkochmanski) [inline lib/cl/READMES/readme-cassandra.lisp]: duplicated sbcl entry, should be `#+(or ccl sbcl)`

File: lib/cl/binary-protocol.lisp [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+    (read-and-decode-integer protocol 2))
+  
+  (defmethod stream-read-i32 ((protocol binary-protocol))
+    (read-and-decode-integer protocol 4))
+
+  (defmethod stream-read-i64 ((protocol binary-protocol))
+    (read-and-decode-integer protocol 8)))
+
+
+(defmethod stream-read-double ((protocol binary-protocol))
```
Comment (dkochmanski) [inline lib/cl/binary-protocol.lisp:100]: remove special code for allegro, we depend on portability library ieee-floats

File: lib/cl/binary-protocol.lisp [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+  (defmethod stream-write-i32 ((protocol binary-protocol) val)
+    (encode-and-write-integer protocol val 4))
+
+  (defmethod stream-write-i64 ((protocol binary-protocol) val)
+    (encode-and-write-integer protocol val 8)))
+
+
+(defmethod stream-write-double ((protocol binary-protocol) val)
+  #+allegro (dolist (b (mapcar #'(lambda (x) (int-bytes x 2))
+                               (multiple-value-list (excl:double-float-to-shorts
```
Comment (dkochmanski) [inline lib/cl/binary-protocol.lisp]: remove allegro (only portability library)

File: lib/cl/conditions.lisp [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+
+(defmethod thrift-error-format-control ((error protocol-error))
+  (concatenate 'string (call-next-method)
+               " connection: ~a."))
+
+(defmethod thrift-error-format-arguments ((error protocol-error))
+  (append (call-next-method)
+          (list (protocol-error-protocol error))))
+
+
```
Comment (dkochmanski) [inline lib/cl/conditions.lisp]: cosmetic: one newline between definitions (if no comment inbetween) is more than enough. remove such gaps.

File: lib/cl/definition-operators.lisp [hunk 6]
```diff
@@ -0,0 +1,545 @@
...
+           ,@(when documentation `(,documentation))
+           (stream-write-message-begin ,gprot ,identifier 'call
+                                       (protocol-next-sequence-number ,gprot))
+           ;; use the respective args structure as a template to generate the message
+           (stream-write-struct ,gprot (thrift:list ,@(mapcar #'(lambda (id name) `(cons ,id ,name)) parameter-ids parameter-names))
+                                ',(str-sym call-struct))
+           (stream-write-message-end ,gprot)
+           ,(if oneway-p
+              nil
+              `(multiple-value-bind (request-message-identifier type sequence)
```
Comment (dkochmanski) [inline lib/cl/definition-operators.lisp]: `,(if oneway-p nil foo)` -> `,(unless oneway-p foo)`

**Review by dkochmanski — CHANGES_REQUESTED** [review]

File: configure.ac (hunk 1 above)
Comment (uint) [reply configure.ac]: I think that would be confusing. The thing is, when you hit `make`, the Common Lisp library will not be built at all, unlike the others. It's only baked into the cross-test binaries when you do `make precross` or `make cross`.

**Review by dkochmanski — APPROVED** [review]

**Pull request merged**